Kudu.Services/GitServer/InfoRefsController.cs

```diff
@@ -185,11 +187,29 @@ public void InitialCommitIfNecessary()
                     return;
                 }
 
-                // do initial commit
-                repository = _repositoryFactory.EnsureRepository(RepositoryType.Git);
+                // if it is a functionApp
+                // git clone will copy repo from wwwroot/ to repository/
+                // then treate it like its not in place deployment
+                if (!string.IsNullOrEmpty(System.Environment.GetEnvironmentVariable(Constants.FunctionRunTimeVersion)))
+                {
+                    // since git clone doesn't happen very often, create this dynamically instead of static
+                    var ignoreDir = new HashSet<string>();
```

Comment:
maybe use a `HashSet<string>(StringComparer.OrdinalIgnoreCase)` is OS is Windows?